### PR TITLE
when creating task with concent enabled, make sure deposit is locked

### LINKED ISSUE #3612: transaction system's `concent_deposit` function doesn't check if the existing deposit is locked
When checking for the existing deposit, we're not ensuring the deposit is in fact locked.

1) When doing my deposit, check if it's locked (in `concent_deposit`)
2) When as provider checking requestor deposit also check if it's locked.

### PR DESCRIPTION
**Pull request by etam**
resolves #3612

### CODE REVIEW

File: golem/task/rpc.py [hunk 1]
```diff
@@ -216,6 +216,7 @@ def _ensure_task_deposit(client, task, force):
             expected=opt_amount,
             force=force,
         )
+        client.transaction_system.concent_relock()
```
Comment (jiivan) [inline golem/task/rpc.py]: This shouldn't be called here.

**Review by jiivan — CHANGES_REQUESTED** [review]

File: golem/task/rpc.py (hunk 1 above)
Comment (etam) [reply golem/task/rpc.py]: Why?

**Comment by etam** [comment]
I've force-pushed a completely new change, so you can safely ignore what was before.

**Review by jiivan — APPROVED** [review]
trisie3

**Pull request merged**